pkg/agent/run.go

```diff
@@ -28,16 +28,30 @@ var AuthToken string
 // Period is the number of seconds between scans
 var Period uint
 
+// Number of times the agent will gather and post data
+var NumberPeriods int
+
 // CredentialsPath is where the agent will try to loads the credentials. (Experimental)
 var CredentialsPath string
 
+
 // Run starts the agent process
 func Run(cmd *cobra.Command, args []string) {
 	ctx := context.Background()
 
-	for {
+	for i := 0; i != NumberPeriods; {
 		gatherAndPostData(ctx)
+		
+		if i == NumberPeriods - 1 {
+			break
+		}
+
 		time.Sleep(time.Duration(Period) * time.Second)
+
+		// Progress loop if a positive number of periods is given
+		if NumberPeriods > 0 {
+			i++
+		}
```

Comment:
I don't think this is necessary, except that it avoids the agent stopping if it runs for long enough to overflow?

Reply:
Yep, if that's not an issue I can remove it :)

Reply:
I think that's not a bad thing to have.

The only reasons not to have it would be:

1) We don't have a counter of the number of loops in case we ever wanted to use that. If we add a counter then that can overflow and maybe cause problems.

2) It's extra lines of code, and fewer lines of code is preferable, all else being equal

`i` is an `int` here, which is either a 32 or 64 bit signed int, depending on the system.
In the worst case of it being a 32 bit signed the max is `2^31 -1`. By my calculations
if we sleep for 1 minute per loop that would mean it would overflow after 4000 years.

Therefore I think it's never really going to happen, and I would err on leaving out
the check in order to have less code, but it's hard to explain that in the code and
that there could be a problem lurking there if the context of this code changes.

Reply:
Good point! In line with this and Charlie's comment I'll switch over to a 'one-shot' flag